File: python/2264_largest_3_same_digit_number_in_string.py
"""
  Problem: 2264 - Largest 3-Same-Digit Number in String  
  Description: Given a string `num` representing a large integer, return the largest good integer as a string. A good integer is a substring of length 3 that consists of the same digit repeated three times (e.g., "777", "000"). If no such substring exists, return an empty string `""`.
  Date: 2025-08-14  
"""

class Solution:
    def largestGoodInteger(self, num: str) -> str:
        for i in range(9,-1, -1):
            pat = str(i) * 3
            if(num.find(pat) != -1) :
                return pat
        
        return ""

File: cpp/2264_largest_3_same_digit_number_in_string.cpp
"""
  Problem: 2264 - Largest 3-Same-Digit Number in String  
  Description: Given a string `num` representing a large integer, return the largest good integer as a string. A good integer is a substring of length 3 that consists of the same digit repeated three times (e.g., "777", "000"). If no such substring exists, return an empty string `""`.
  Date: 2025-08-14  
"""

class Solution {
public:
    string largestGoodInteger(string num) {
        int target = -1;
        for(int i = 0; i < 10 ; i++){
            string pattren(3,'0' + i);
            if (num.find(pattren) != string::npos){
                target = i;
            }
        }
        if (target == -1){
            return "";
        } else{
            string pattren(3,'0' + target);
            return pattren;
        }
        
    }
};
